### PR TITLE
Added feature flag to use the new geocoder metrics system or the legacy system

### LINKED ISSUE #6411: Migrate quota check for geocoding logic to use redis instead of database when the geocoding quota is calculated
Name of the feature flag **new_geocoder_quota**

- [x] Create feature flag to switch between the legacy and the new geocoder metrics system

### PR DESCRIPTION
**Pull request by ethervoid**
Closes #6411 

This PR add the possibility to use either the new or the legacy system based on the 'new_geocoder_quota' feature flag.

### CODE REVIEW

**Comment by ethervoid** [comment]
retest this please

**Comment by ethervoid** [comment]
@rafatower check this out please

File: spec/models/organization_shared_examples.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@
       expect {
         ::User.any_instance.expects(:get_geocoding_calls).never
         get_geocoding_calls_by_organization_id(@organization.id).should == 14
-      }.to make_database_queries(count: 0..1)
+      }.to make_database_queries(count: 0..5)
```
Comment (rafatower) [inline spec/models/organization_shared_examples.rb]: just curious: why did this change?

**Comment by rafatower** [comment]
:+1:

File: spec/models/organization_shared_examples.rb (hunk 1 above)
Comment (ethervoid) [inline spec/models/organization_shared_examples.rb]: Due the use of a feature flag the number of queries is 5. 1 for the original purpose and 4 to get the feature flag info

**Pull request merged**